services/wfm/src/main/java/org/openkilda/wfm/topology/flow/validation/FlowValidator.java

```diff
@@ -68,6 +71,34 @@ void checkBandwidth(Flow flow) throws FlowValidationException {
         }
     }
 
+    /**
+     * Checks a flow for conflicts with ISL ports.
+     *
+     * @param requestedFlow a flow to be validated.
+     * @throws FlowValidationException is thrown in a case when flow endpoints conflict with existing ISL ports.
+     */
+    void checkFlowForIslConflicts(Flow requestedFlow) throws FlowValidationException {
+        // Check the source
+        if (!islRepository.findBySrcEndpoint(requestedFlow.getSrcSwitch().getSwitchId(),
```

Comment:
It would be more optimal to create a combined method in IslRepository to lookup an isl by both source or destination. So we'll have a single call to Neo4j instead of 2.

Reply:
You are right. Fixed.